components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -256,8 +256,14 @@ public void handle(HttpServletRequest request, HttpServletResponse response) thr
                         }
                         diagnosticLog.info("Restarting the authentication flow from step 1 for  " +
                                 context.getContextIdentifier());
-                        context.setCurrentStep(1);
+                        context.setCurrentStep(0);
                         context.setProperty(BACK_TO_FIRST_STEP, true);
+                        Map<String, String> runtimeParams = context
+                                .getAuthenticatorParams(FrameworkConstants.JSAttributes.JS_COMMON_OPTIONS);
+                        runtimeParams.put(FrameworkConstants.JSAttributes.JS_OPTIONS_USERNAME, null);
```

Comment:
let's remove all the authenticator params. let's not specifically remove the username. As we are restarting the flow, we can remove all the authenticator params

Reply:
fixed in e1b812d